src/Altinn.Platform/Altinn.Platform.Profile/Profile/Services/Decorators/UserProfileCachingDecorator.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Threading.Tasks;
+
+using Altinn.Platform.Profile.Configuration;
+using Altinn.Platform.Profile.Models;
+using Altinn.Platform.Profile.Services.Interfaces;
+
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Profile.Services.Decorators
+{
+    /// <summary>.
+    /// Decorates an implementation of IUserProfiles by caching the userProfile object.
+    /// If available, object is retrieved from cache without calling the service
+    /// </summary>
+    public class UserProfileCachingDecorator : IUserProfiles
+    {
+        private readonly IUserProfiles _decoratedService;
+        private readonly IMemoryCache _memoryCache;
+        private readonly MemoryCacheEntryOptions _cacheOptions;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserProfileCachingDecorator"/> class.
+        /// </summary>
+        /// <param name="decoratedService">The decorated userProfiles service</param>
+        /// <param name="memoryCache">The memory cache</param>
+        /// <param name="generalSettings">The general settings</param>
+        public UserProfileCachingDecorator(
+            IUserProfiles decoratedService,
+            IMemoryCache memoryCache,
+            IOptions<GeneralSettings> generalSettings)
```

Comment:
Now that you have split the responsibilities of the http client and the caching into separate classes you could move the ProfileCacheLifetimeSeconds into it's own CacheSettings and by that reduce dependencies to a minimum for both. The http client doesn't need to know about cache settings and vice versa.

Reply:
Thinking that might be a bit of over engineering for a component like Profile. That would result in three different setting files each containing one or two properties. Seems like something to consider if the Platform Profile should grow in size or include a larger set of configuraitons.

Reply:
Not different config files (if that's what you mean?) - they all live in the same config file just different sections. Then you use the IOption pattern to map the section to their own class (but thats perhaps what you mean by multiple files?). The point being that you group together what belongs together and you minimize the indirect dependencies that a class has. Because by taking the GeneralSettings as a dependency into the caching decorator your basically saying that you need the whole interface for this class to work - but you don't, you only need the cache setting. And as we keep adding settings to the general settings the indirect dependency gets bigger and bigger. And small interfaces are way easier to handle and test (you won't even need Moq), and multiple classes is a small price to pay imo. And if you would like to avoid multiple class files you can have one, but implement mulitple interfaces in the same class. But maybe next time :-)